app/src/main/java/org/schabi/newpipe/about/LicenseFragmentHelper.kt

```diff
@@ -67,11 +67,10 @@ private fun getHexRGBColor(context: Context, color: Int): String {
 }
 
 fun showLicense(context: Context?, component: SoftwareComponent): Disposable {
-    return showLicense(context, component.license) { alertDialog ->
-        alertDialog.setPositiveButton(R.string.dismiss) { dialog, _ ->
+    return showLicense(context, component.license) {
+        setPositiveButton(R.string.dismiss) { dialog, _ ->
             dialog.dismiss()
-        }
-        alertDialog.setNeutralButton(R.string.open_website_license) { _, _ ->
+        }.setNeutralButton(R.string.open_website_license) { _, _ ->
```

Comment:
I would keep this on a new line, IMO.

Reply:
> Also, I don't believe the default parameter commit is particularly useful.

It allows the method overload to be removed.

Reply:
I'm aware, it's just that IMO it didn't really do anything useful here, considering that both `showLicense` have different block arguments for the "backend" `showLicense`'s AlertDialog. I think the previous one is simpler to read, even if it makes the class have a few more lines. If it's merged as is, then the block's default parameter (with its arguments) is just overridden by the block arguments set in `showLicense(Context, SoftwareComponent)`. I do understand the reasoning behind it, I just think that it makes it overall harder to interpret.

I can't explain very well right now, my brain is kinda fried.